packages/paste-core/components/popover/src/PopoverButton.tsx

```diff
@@ -1,26 +1,40 @@
 import * as React from 'react';
 import * as PropTypes from 'prop-types';
-import type {ButtonProps} from '@twilio-paste/button';
-import type {BoxProps} from '@twilio-paste/box';
 import {Button} from '@twilio-paste/button';
+import {Badge} from '@twilio-paste/badge';
 import {NonModalDialogDisclosurePrimitive} from '@twilio-paste/non-modal-dialog-primitive';
+import type {PopoverButtonProps, ButtonBadgeProps} from './types';
 import {PopoverContext} from './PopoverContext';
 
-export interface PopoverButtonProps extends ButtonProps, Pick<BoxProps, 'element'> {
-  id?: string;
-  children: React.ReactNode;
-  toggle?: () => void;
-}
+// eslint-disable-next-line react/display-name
+const ButtonBadge = React.forwardRef<HTMLButtonElement, ButtonBadgeProps>(({children, ...props}, ref) => (
+  <Badge {...props} as="button" ref={ref}>
+    {children}
+  </Badge>
+));
 
 const PopoverButton = React.forwardRef<HTMLButtonElement, PopoverButtonProps>(
-  ({children, element = 'POPOVER_BUTTON', ...popoverButtonProps}, ref) => {
+  ({as = 'button', children, element = 'POPOVER_BUTTON', ...popoverButtonProps}, ref) => {
```

Comment:
This is fair.... I talked to @SiTaggart about this and he suggested using `as` this way because it scales better if we ever want to add more options for PopoverButton.

Reply:
If we change it to `as={Badge}`, how can we type it/add checks that they only pass either `Button` or `Badge`?

Also, for badge right now, I have to pass the `as="button"` prop like this:
```
const ButtonBadge = React.forwardRef<HTMLButtonElement, ButtonBadgeProps>(({children, ...props}, ref) => (
  <Badge {...props} as="button" ref={ref}>
    {children}
  </Badge>
));
```

How would we do that if you pass `as={Badge}`?

Reply:
Yeah, ok, that is a fair valid point. `as="a" or as={Button}` is very default Emotion behavior, and adding a new, kind custom string matching behaviour to that API might be a stretch too far.

As for doing something like `as={Badge}`, it's possible, but to get the typings right based on the Component you pass it you have to get pretty deep into Polymorphic types, and I'm just not sure I'm up to it yet. It's gnarly. One day, yup. Purhaps, not today. (reading: https://www.benmvp.com/blog/polymorphic-react-components-typescript/ and https://github.com/kripod/react-polymorphic-types).

So I guess that leaves `<PopoverBadgeButton />` as the least worst for now.

I don't love it, because we might end up with a whole bunch of them in the future, but, in absence of a sensible alternative for right now... it's good enough?